custom-elements/custom-elements-registry/when-defined.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<title>Custom Elements: CustomElementsRegistry.whenDefined</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<div id="log"></div>
+<iframe id="iframe"></iframe>
+<script>
+'use strict';
+(() => {
+  // https://html.spec.whatwg.org/multipage/scripting.html#dom-customelementsregistry-whendefined
+
+  // Use window from iframe to isolate the test.
+  const testWindow = iframe.contentDocument.defaultView;
+  const customElements = testWindow.customElements;
+
+  let testable = false;
+  test(() =>  {
+    assert_true('customElements' in testWindow, '"window.customElements" exists');
+    assert_true('whenDefined' in customElements, '"window.customElements.whenDefined" exists');
+    testable = true;
+  }, '"window.customElements.whenDefined" should exist');
+  if (!testable)
+    return;
+
+  // 1. If name is not a valid custom element name,
+  // then return a new promise rejected with a "SyntaxError" DOMException and abort these steps.
+  // TODO: should share invalidCustomElementNames from defined.html.
+
+  // 2. If this CustomElementsRegistry contains an entry with name name,
+  // then return a new promise resolved with undefined and abort these steps.
+  test(() => {
+    const name = 'test-when-defined-existing';
+    customElements.define(name, class {});
+    let resolved, resolvedWith;
+    customElements.whenDefined(name)
+      .then(result => { resolved = true; resolvedWith = result; });
+    assert_true(resolved, 'whenDefined for existing name should resolve immediately');
+    assert_equals(resolvedWith, undefined, 'whenDefined should resolve with undefined');
+  }, 'whenDefined for existing name');
```

Comment:
This will always fail because `resolved` is not `true` until the promise resolves, which happens async.

You should use `promise_test` for this. Something like

```js
promise_test(() => {
  const name = 'test-when-defined-existing';
  customElements.define(name, class {});

  return customElements.whenDefined(name).then(value => {
    assert_equals(value, undefined, 'The promise should fulfill with undefined');
  });
});
```